### PR TITLE
[REF-6276] Added component to save dataframe to file-name provided

### PR DESCRIPTION
**Pull request by PMjoydas**
Added component to save dataframe to file

* Tested the component running in mlpiper and MLApp

### CODE REVIEW

File: components/Python/Scikit-learn/dataframe_to_file/component.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      "defaultComponent": "",
+      "type": "str",
+      "group": "data"
+    }
+  ],
+  "group": "Sinks",
+  "arguments": [
+    {
+      "key": "sinkfilename",
+      "label": "save to file",
```
Comment (yakov-g) [inline components/Python/Scikit-learn/dataframe_to_file/component.json:35]: probably it should be "FIlename"

File: components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    Adapter for write_df_to_file 
+    """
+
+    def __init__(self, engine):
+        super(self.__class__, self).__init__(engine)
+
+    def _materialize(self, parent_data_objs, user_data):
+        file_path = self._params["sinkfilename"]
+        df_data = parent_data_objs[0]
+        return[write_df_to_file(df_data, file_path)]
```
Comment (yakov-g) [inline components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py]: space after `return`

File: components/Python/Scikit-learn/dataframe_to_file/component.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "engineType": "Generic",
+  "language": "Python",
+  "userStandalone": false,
+  "name": "dataframe_to_file",
+  "label": "Sink DataFrame to File",
+  "program": "dataframe_to_file.py",
```
Comment (yakov-g) [inline components/Python/Scikit-learn/dataframe_to_file/component.json:10]: add description that the component writes DF to file, and passes filename to the next component

File: components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def write_df_to_file(df_data, filepath):
+    """
+    Save DataFrame to file
+    """
+    suffix_time_stamp = str(int(time.time()))
+    save_file = str(filepath) + '.' + suffix_time_stamp
+    sfile = open(save_file, 'w+')
+    pandas.DataFrame(df_data).to_csv(save_file)
```
Comment (yakov-g) [inline components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py]: According to docs, first param of the to_csv func is:
```path_or_buf : str or file handle, default None
File path or object, if None is provided the result is returned as a string. If a file object is passed it should be opened with newline=’‘, disabling universal newlines.

Changed in version 0.24.0: Was previously named “path” for Series.```

you open a file, but pass a path.... 
As I undestand, you can to pass path; or open and pass the object. Check please.

Comment (PMjoydas) [reply components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py]: $ pip freeze | grep pandas
pandas==0.24.2

parameter to "to_csv" is string so it is valid, however I will add "path_or_buf=" so as its clear specifically for the release post 0.24.0

File: components/Python/Scikit-learn/dataframe_to_file/component.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  "arguments": [
+    {
+      "key": "filename",
+      "label": "save to file",
+      "type": "str",
+      "description": "File used to save DataFrame contents",
+      "optional": true
+    }
+  ],
+  "version": 1
```
Comment (liororama) [inline components/Python/Scikit-learn/dataframe_to_file/component.json]: Move version to the top, also group

File: components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    def __init__(self, engine):
+        super(self.__class__, self).__init__(engine)
+
+    def _materialize(self, parent_data_objs, user_data):
+        file_path = self._params["sinkfilename"]
+        df_data = parent_data_objs[0]
+        return [write_df_to_file(df_data, file_path)]
+
+
+def write_df_to_file(df_data, filepath):
```
Comment (liororama) [inline components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py]: move this function to be inside the class

File: components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    """
+    Save DataFrame to file
+    """
+    suffix_time_stamp = str(int(time.time()))
+    save_file = str(filepath) + '.' + suffix_time_stamp
+    sfile = open(save_file, 'w+')
```
Comment (liororama) [inline components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py]: Better to write
with open(...) as sfile:
   xxxx

Then no need for the close

File: components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    pandas.DataFrame(df_data).to_csv(path_or_buf=sfile, encoding='utf-8')
+    sfile.close()
+    if not os.path.exists(save_file):
+        self._logger.info("stderr- failed to write {}".format(save_file), file=sys.stderr)
```
Comment (liororama) [inline components/Python/Scikit-learn/dataframe_to_file/dataframe_to_file.py]: Why .info when it is an error?

**Review by yakov-g — APPROVED** [review]

**Pull request merged**